extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -387,12 +387,95 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 				},
 				recommendationError: error
 			};
-		}
+		}		// Generating all the telemetry asynchronously as we don't need to block the user for it.
+		this.generateSkuRecommendationTelemetry().catch(e => console.error(e));
 
-		// this.generateAssessmentTelemetry().catch(e => console.error(e));
 		return this._skuRecommendationResults;
 	}
 
+	private async generateSkuRecommendationTelemetry(): Promise<void> {
+		try {
+
+			this._skuRecommendationResults?.recommendations?.sqlMiRecommendationResults?.forEach(resultItem => {
+				// Send telemetry for recommended MI SKU
+				sendSqlMigrationActionEvent(
+					TelemetryViews.SkuRecommendationWizard,
+					TelemetryAction.GetMISkuRecommendation,
+					{
+						'sessionId': this._sessionId,
+						'RecommendedSku': JSON.stringify(resultItem?.targetSku)
+					},
+					{}
+				);
+			});
+
+			this._skuRecommendationResults?.recommendations?.sqlVmRecommendationResults?.forEach(resultItem => {
+				// Send telemetry for recommended VM SKU
+				sendSqlMigrationActionEvent(
+					TelemetryViews.SkuRecommendationWizard,
+					TelemetryAction.GetVMSkuRecommendation,
+					{
+						'sessionId': this._sessionId,
+						'RecommendedSku': JSON.stringify(resultItem?.targetSku)
+					},
+					{}
+				);
+			});
+
+			// Send Instance requirements used for calculating recommendations
+			sendSqlMigrationActionEvent(
+				TelemetryViews.SkuRecommendationWizard,
+				TelemetryAction.GetInstanceRequirements,
+				{
```

Comment:
can we add the performance data source to track if user was getting the recommendation using "collect data" or "open existing"?
`performanceDataSource = this._skuRecommendationPerformanceDataSource`